File: ts/example/modules/greetings/greetings_builder.ts
import { ModuleBuilder } from '../../../src/index';
import { HelloInput, HelloOutput, HelloWorld } from './usecases/hello_world';

export function buildGreetingsModule(m: ModuleBuilder): void {
  m.usecase('hello', HelloWorld.fromJson, {
    inputClass: HelloInput,
    outputClass: HelloOutput,
  });
}

File: dart/example/modules/greetings/greetings_builder.dart
import 'package:modular_api/modular_api.dart';

import 'usecases/hello_world.dart';

void buildGreetingsModule(ModuleBuilder m) {
  m.usecase(
    'hello',
    HelloWorld.fromJson,
    inputExample: HelloInput.example,
    outputExample: HelloOutput.example,
  );
}